// @unvt/charites/dist/lib/defaultValues.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.defaultSettings = exports.defaultValues = void 0;
const path_1 = __importDefault(require("path"));
const os_1 = __importDefault(require("os"));
const fs_1 = __importDefault(require("fs"));
const js_yaml_1 = __importDefault(require("js-yaml"));
const homedir = os_1.default.homedir();
const defaultProvider = "default";
const configDir = path_1.default.join(homedir, '.charites');
fs_1.default.mkdirSync(configDir, { recursive: true });
const configFile = path_1.default.join(configDir, 'config.yml');
let config = { provider: '', providerDir: '', mapboxAccessToken: '' };
try {
    const yaml = fs_1.default.readFileSync(configFile, 'utf-8');
    config = js_yaml_1.default.load(yaml);
}
catch (e) {
    // nothing to do
}
exports.defaultValues = {
    provider: config.provider || defaultProvider,
    providerDir: path_1.default.join(path_1.default.dirname(path_1.default.dirname(__dirname)), 'provider'),
    mapboxAccessToken: config.mapboxAccessToken || ''
};
exports.defaultSettings = {
    configFile: configFile
};
